src/validation/__tests__/KnownDirectivesRule-test.ts

```diff
@@ -219,9 +222,13 @@ describe('Validate: Known directives', () => {
         message: 'Directive "@onQuery" may not be used on FIELD.',
         locations: [{ column: 19, line: 16 }],
       },
+      {
+        message: 'Directive "@onField" may not be used on VARIABLE_DEFINITION.',
```

Comment:
Not blocking for this PR, but we may want to consider this directive location name being different - maybe to `FRAGMENT_VARIABLE_DEFINITION`?

Reply:
Yeah it's weird: if they're their own AST node then absolutely. If not... maybe? I think probably?

This is a small change and another thing I flip-flopped on.

Reply:
DirectiveLocations and AST nodes are not a perfect 1:1

See `getDirectiveLocationForASTPath()` for the mapping, there's already a few bits of logic there to consider extending in a similar way.

I think keeping the AST minimally changed is great, so it makes sense to me to have the `VARIABLE_DEFINITION` AST node map to one of two directive locations based on the parent AST node being a fragment or not.